app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -89,6 +93,34 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  def fetchLogData(appId: String, attemptId: Option[String])(
```

Comment:
I think there is a design issue here. First, it is inconsistent that all the other methods that are getting data via rest (getApplicationInfo, getJobDatas etc. ) are defined as private methods. And, `fetchLogData` is a public method. The other problem is that on line 99 does not consider attemptId while generating appTarget, which is incorrect (please see line 82).

I think a better design would be to introduce an optional member, `restEventLogData`, in class `SparkRestDerivedData` and populate that member in the class. Define a private method, `getEventLogData`, in this class which directly returns Optional[SparkLogDerivedData]. The method `getEventLogData` can take attemptTarget (defined on line 82) as one of the arguments, in addition to appId, and attemptId. This seems more consistent with rest of the code, and you also don't need to recreate `appTarget`.